sourced/ml/models/id_splitter.py

```diff
@@ -0,0 +1,124 @@
+    VENDOR = "source{d}"
+    DESCRIPTION = "Model that contains source code identifier splitter BiLSTM weights."
+    LICENSE = DEFAULT_LICENSE
+
+    def construct(self, model: "keras.models.Model" = None,
+                  session: "tf.Session" = None):
+        assert model is not None
+
+        if session is not None:
+            # config = tf.ConfigProto()
+            # config.gpu_options.allow_growth = True
+            # tf_session = tf.Session(config=config)
+            # backend.tensorflow_backend.set_session(tf_session)
+            # else:
+            backend.tensorflow_backend.set_session(session)
+        os.environ["TF_CPP_MIN_LOG_LEVEL"] = "1"
+
+        self._model = model
+        return self
+
+    @property
+    def model(self):
+        """
+        Returns the wrapped keras model.
+        """
+        return self._model
+
+    def clear_keras(self):
+        """
+        Clears the keras session used to run the model
+        """
+        keras.backend.clear_session()
+
+    def _generate_tree(self) -> dict:
+        return {
+                "config": self._model.get_config(),
+                "weights": self._model.get_weights()
+                }
+
+    def _load_tree(self, tree: dict):
+        from keras.models import Model
+
+        model = Model.from_config(tree["config"])
+        model.set_weights(tree["weights"])
+        self.construct(model=model)
+
+    def _prepare_single_identifier(self, identifier, maxlen=MAXLEN, padding=PADDING, mapping=None):
+        if mapping is None:
+            mapping = dict((c, i + 1) for i, c in enumerate(string.ascii_lowercase))
+
+        # Clean identifier
+        clean_id = "".join([char for char in identifier.lower() if char in string.ascii_lowercase])
+        if len(clean_id) > MAXLEN:
+            clean_id = clean_id[:MAXLEN]
+        logging.info("Preprocessed identifier: {}".format(clean_id))
+        return np.array([mapping[c] for c in clean_id], dtype="int8"), clean_id
+
+    def prepare_input(self, identifiers: [str], maxlen: int = MAXLEN, padding: int = PADDING,
+                      mapping=None):
+
+        processed_ids = []
+        clean_ids = []
+        for identifier in identifiers:
+            feat, clean_id = self._prepare_single_identifier(
+                identifier, maxlen=maxlen, padding=padding)
+            processed_ids.append(feat)
+            clean_ids.append(clean_id)
+
+        processed_ids = pad_sequences(processed_ids, maxlen=maxlen,
+                                      padding=padding)
+
+        return processed_ids, clean_ids
+
+    def load_model_file(self, path="./id_splitter_rnn.model"):
+
+        self._model = keras.models.load_model(path, custom_objects={"precision": precision,
+                                                                    "recall": recall,
+                                                                    "f1score": f1score})
+
+    def __call__(self, tokens: [str]) -> [[str]]:
```

Comment:
```suggestion
    def __call__(self, tokens: Sequence[str]) -> List[str]:
```
please refer to https://docs.python.org/3/library/typing.html